metals/src/main/scala/scala/meta/internal/metals/MetalsPasteProvider.scala

```diff
@@ -0,0 +1,107 @@
+package scala.meta.internal.metals
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codeactions.MissingSymbolDiagnostic
+import scala.meta.internal.semanticdb.Scala.Descriptor.Method
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j
+import org.eclipse.lsp4j.TextDocumentPositionParams
+import org.eclipse.lsp4j.TextEdit
+
+class MetalsPasteProvider(
+    compilers: Compilers,
+    buildTargets: BuildTargets,
+)(implicit ec: ExecutionContext) {
+
+  def didPaste(
+      params: MetalsPasteParams,
+      cancelToken: CancelToken,
+  ): Future[List[TextEdit]] = {
+    val path = params.textDocument.getUri.toAbsolutePath
+    val isScala3 =
+      buildTargets.scalaVersion(path).exists(ScalaVersions.isScala3Version)
+    val MissingSymbol = new MissingSymbolDiagnostic(isScala3)
+
+    compilers.diagnostics(path, cancelToken).flatMap { diagnostics =>
+      val edits = diagnostics.collect {
+        case d @ MissingSymbol(name, findExtensionMethods)
+            if params.range.overlapsWith(d.getRange()) =>
+
+          val offset =
+            if (isScala3) d.getRange().getEnd()
+            else d.getRange().getStart()
+
+          val importPositionParams =
+            new TextDocumentPositionParams(params.textDocument, offset)
+          val symbolsPositionParams =
+            new TextDocumentPositionParams(
+              params.originDocument,
+              adjustPositionToOrigin(params, offset),
+            )
+          for {
+            imports <- compilers.autoImports(
+              importPositionParams,
+              name,
+              findExtensionMethods,
+              cancelToken,
+            )
+            symbols <- compilers.symbolsAt(symbolsPositionParams, cancelToken)
```

Comment:
Could we instead do .definition on the original symbol to get the exact thing to import? We wouldn't need to add a new method and it would work in some weird scenarios also (although I can't think of exact problematic now currently). Would need to just do the definition on `orginalFileOffset + currentOffset - currentStartOffset`